### PR TITLE
ARROW-13700: [Docs][C++] Clarify DayOfWeekOptions args

### PR DESCRIPTION
**Pull request by rok**
This is to resolve [ARROW-13700](https://issues.apache.org/jira/browse/ARROW-13700).

### CODE REVIEW

**Comment by rok** [comment]
@ianmcook

**Comment by rok** [comment]
@lidavidm - could I ask you for a quick doc check?

File: cpp/src/arrow/compute/api_scalar.h [hunk 1]
```diff
@@ -323,7 +323,8 @@ struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
 
   /// Number days from 0 if true and from 1 if false
   bool count_from_zero;
-  /// What day does the week start with (Monday=1, Sunday=7)
+  /// What day does the week start with (Monday=1, Sunday=7). This is unaffected
+  /// by the count_from_zero parameter.
   uint32_t week_start;
```
Comment (lidavidm) [inline cpp/src/arrow/compute/api_scalar.h:328]: Perhaps:

```
  /// Number output days from 0 if true and from 1 if false
  bool count_from_zero;
  /// What day does the week start with (Monday=1, Sunday=7).
  /// The numbering is unaffected by the count_from_zero parameter.
  uint32_t week_start;
```

**Review by lidavidm — APPROVED** [review]
LGTM, maybe one wording suggestion to clarify.

File: cpp/src/arrow/compute/api_scalar.h (hunk 1 above)
Comment (rok) [reply cpp/src/arrow/compute/api_scalar.h:328]: Yeah, that makes sense.

Comment (rok) [reply cpp/src/arrow/compute/api_scalar.h:328]: Changed.

**Review by lidavidm — APPROVED** [review]

**Pull request closed without merging**